src/javascript-extract.js

```diff
@@ -108,8 +73,59 @@ function getGettextEntriesFromJavaScript(script, parser = 'auto') {
   return extractedEntries;
 }
 
+function parseJSGettextWithAcorn(script) {
+  let allTokens = [];
+
+  const ACORN_OPTIONS = {
+    ecmaVersion: 10,
+    sourceType: 'module',
+    locations: true,
+    onToken: allTokens,
+    plugins: {
+      stage3: true,
+    },
+  };
+
+  Parser.extend(stage3).parse(script, ACORN_OPTIONS);
+
+  return allTokens;
+}
+
+function parseJSGettextWithBabel(script) {
+  let allTokens = [];
+
+  const babelResult = babel.parseSync(script, {
+    sourceType: 'module',
+    parserOpts: {
+      tokens: true,
+    },
+  });
+  allTokens.push(...babelResult.tokens);
+
+  return allTokens;
+}
+
 function extractStringsFromJavascript(filename, script, parser = 'auto') {
-  return getTextEntries(filename, getGettextEntriesFromJavaScript(script, parser));
+  let extractedEntries;
+
+  switch (parser) {
+  case 'acorn':
+    extractedEntries = getGettextEntriesFromJavaScript(parseJSGettextWithAcorn(script));
+    break;
+  case 'babel':
+    extractedEntries = getGettextEntriesFromJavaScript(parseJSGettextWithBabel(script));
+    break;
+
+  default:
+    try {
+      extractedEntries = getGettextEntriesFromJavaScript(parseJSGettextWithAcorn(script));
+    } catch (e) {
+      console.log(`[${PROGRAM_NAME}] will switch extracting using acorn as parser and use babel instead`); // eslint-disable-line no-console
+      extractedEntries = getGettextEntriesFromJavaScript(parseJSGettextWithBabel(script));
```

Comment:
seems to me that you could use `getGettextEntriesFromJavaScript` only once at the end of the function, whoich would make it way clearer.

Reply:
Ah, smart idea! Implemented now